xds/src/test/java/io/grpc/xds/LocalityStoreTest.java

```diff
@@ -842,6 +865,393 @@ public void reset() {
+   * EDS update, localities moved: P0 sz1, sz3; P1 sz4; P2 sz2 - P0 C, P1 R, P2 R&D
+   * 15 min passes - P0 C, P1 R, P2 N/A
+   * EDS update, locality removed: P0 sz1, sz3, - P0 C, P1 N/A, sz4 R&D
+   * sz3 gets READY - P0 R, P1 N/A, sz4 R&D
+   * EDS update, locality comes back and another removed: P0 sz1, P1 sz4 - P0 C, P1 R, sz3 R&D
+   */
+  @Test
+  public void multipriority() {
+    // EDS update: P0 sz1; P1 sz2; P2 sz3; P3 sz4 - P0 C, P1 N/A, P2 N/A, P3 N/A
+    LocalityInfo localityInfo1 =
+        new LocalityInfo(ImmutableList.of(lbEndpoint11, lbEndpoint12), 1, 0);
+    LocalityInfo localityInfo2 =
+        new LocalityInfo(ImmutableList.of(lbEndpoint21, lbEndpoint22), 2, 1);
+    LocalityInfo localityInfo3 =
+        new LocalityInfo(ImmutableList.of(lbEndpoint31, lbEndpoint32), 3, 2);
+    LocalityInfo localityInfo4 =
+        new LocalityInfo(ImmutableList.of(lbEndpoint41, lbEndpoint42), 3, 3);
+    final Map<XdsLocality, LocalityInfo> localityInfoMap = ImmutableMap.of(
+        locality1, localityInfo1, locality2, localityInfo2, locality3, localityInfo3, locality4,
+        localityInfo4);
+    syncContext.execute(new Runnable() {
+      @Override
+      public void run() {
+        localityStore.updateLocalityStore(localityInfoMap);
+      }
+    });
+    assertThat(loadBalancers.keySet()).containsExactly("sz1");
+    LoadBalancer lb1 = loadBalancers.get("sz1");
+    InOrder inOrder = inOrder(lb1, helper);
+    ArgumentCaptor<ResolvedAddresses> resolvedAddressesCaptor1 =
+        ArgumentCaptor.forClass(ResolvedAddresses.class);
+    inOrder.verify(helper).updateBalancingState(CONNECTING, BUFFER_PICKER);
+    inOrder.verify(lb1).handleResolvedAddresses(resolvedAddressesCaptor1.capture());
+    assertThat(resolvedAddressesCaptor1.getValue().getAddresses()).containsExactly(eag11, eag12);
+
+    // 10 sec passes P0 still CONNECTING - P0 C, P1 C, P2 N/A, P3 N/A
+    fakeClock.forwardTime(9, TimeUnit.SECONDS);
+    assertThat(loadBalancers.keySet()).containsExactly("sz1");
+    assertThat(fakeClock.getPendingTasks(failOverTaskFilter)).hasSize(1);
+    ScheduledTask failOverTask =
+        Iterables.getOnlyElement(fakeClock.getPendingTasks(failOverTaskFilter));
+    fakeClock.forwardTime(1, TimeUnit.SECONDS);
+    assertThat(failOverTask.isCancelled()).isFalse();
+    assertThat(fakeClock.getPendingTasks(failOverTaskFilter)).doesNotContain(failOverTask);
+    assertThat(fakeClock.getPendingTasks(failOverTaskFilter)).hasSize(1);
+    failOverTask = Iterables.getOnlyElement(fakeClock.getPendingTasks(failOverTaskFilter));
+    assertThat(fakeClock.getPendingTasks(deactivationTaskFilter)).isEmpty();
+    assertThat(loadBalancers.keySet()).containsExactly("sz1", "sz2");
+    LoadBalancer lb2 = loadBalancers.get("sz2");
+    inOrder = inOrder(lb1, lb2, helper);
+    inOrder.verify(helper).updateBalancingState(CONNECTING, BUFFER_PICKER);
+    ArgumentCaptor<ResolvedAddresses> resolvedAddressesCaptor2 =
```

Comment:
nit: if you prefer using `ArgumentCaptor.forClass(null)`, use it everywhere, but not a mix of both styles.

Reply:
It's already mixed in this class :(